packages/styled-components/src/models/StyledComponent.ts

```diff
@@ -143,7 +143,7 @@ function useStyledComponentImpl<Props extends object>(
     if (context[key] === undefined) {
       // Omit undefined values from props passed to wrapped element.
       // This enables using .attrs() to remove props, for example.
-    } else if (key[0] === '$' || key === 'as' || key === 'theme') {
```

Comment:
I'd rather not do this, can use `useTheme` to get the theme in your component if needed.

Reply:
Sorry, I didn't know how.
Could you please show me an example?

Reply:
https://styled-components.com/docs/api#usetheme

Reply:
@probablyup I think there is misunderstanding about the intention of this change. 
It is not about to acquire the theme via the theme prop or via the useTheme hook. It is not about the styled-components theming mechanism at all. 
It is about other libs relying on the "theme" prop which is now "killed" by the styled-components wrapper.

Many internal components of react-select passing "theme" props around. When you style them via styled-components, this prop is gone and we ran into a runtime error.
This was also mentioned in #4074.

So changing the line 146 to not remove the "theme" prop fixes that issue with react-select.

Reply:
@quantizor Could we please have this PR added or implement other change to fix the https://github.com/styled-components/styled-components/issues/4074. issue due to which it is impossible to use react-select with styled components?

Reply:
I see. Yeah we can fix this.

Reply:
The only thing I'm not sure about is what should happen if a theme prop is passed that is incompatible with the styled-components theme. The styles still need the context theme to work, whereas the theme being described here has nothing to do with styled-components.

Reply:
Maybe the theme should be a second argument to style interpolations instead of being lumped in with all props. That would disambiguate the two, but we wouldn't be able to remove the s-c theme from the props object until v7 since it would be a breaking change.

Reply:
Further complicating this, we allow the s-c theme to be overridden _via theme prop_ as a general feature...

Reply:
@quantizor Not sure that I understand in details what was changed in s-c v6 but it worked in v5. Here is the [sandbox](https://codesandbox.io/p/sandbox/codesandboxer-example-forked-l7s2ny?file=%2Fexample.tsx%3A18%2C6). If you downgrade s-c to  5.3.11 there is no any error.